compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -266,6 +266,27 @@ fn link_rlib<'a, B: ArchiveBuilder<'a>>(
 
     let mut ab = <B as ArchiveBuilder>::new(sess, out_filename, None);
 
+    let metadata = match flavor {
```

Comment:
Nit:

```suggestion
    let trailing_metadata = match flavor {
```

perhaps? plain `metadata` seems overly generic for the somewhat nontrivial logic happening here.